docs/docsite/rst/dev_guidelines.rst

```diff
@@ -172,6 +172,20 @@ authentication parameters.  To do the same for your new module, add an entry for
   aws_module_name:
   - aws
 
+Module behavior
+---------------
+
+To reduce the chance of breaking changes occurring when new features are added,
+when a parameter is not explicitly set in a task, the module should avoid
+modifying the resource attribute.
+
+By convention, when a parameter is explicitly set in a task, the module should
+set the resource attribute to match what was set in the task.  In some cases,
+such as tags or associations, it can be helpful to add an additional parameter
+which changes this behavior to add the new setting to the attribute rather than
+replacing the current attributes.  Such a parameter should default to the
+replacive rather than additive behavior.
+
```

Comment:
An additional parameter to change the behaviour so it would be additive.  An example being `purge_tags` which should default to `True` (replacive), but can be set to `False` to make the tag behaviour additive instead.

Reply:
@samccann I've tried rewording this...

Reply:
yeah that's better, thanks!